server/session/ratelimiter.go

```diff
@@ -1,93 +1,65 @@
 package session
 
 import (
-	"fmt"
-	"time"
+	"sync"
 
 	"github.com/argoproj/argo-cd/util/session"
 
 	util "github.com/argoproj/gitops-engine/pkg/utils/io"
-	"github.com/bsm/redislock"
-	"github.com/go-redis/redis"
 	log "github.com/sirupsen/logrus"
 )
 
-const (
-	lockKey                = "login:lock"
-	inProgressCountKey     = "login:in-progress-count"
-	inProgressTimeoutDelay = time.Minute
-)
+var once sync.Once
 
-type stateStorage interface {
-	obtainLock(key string, ttl time.Duration) (util.Closer, error)
-	set(key string, value interface{}, expiration time.Duration) error
-	get(key string) (int, error)
-}
+const lockKey = "login:lock"
 
-// NewRedisStateStorage creates storage which leverages redis to establish distributed lock and store number
-// of incomplete login requests.
-func NewRedisStateStorage(client *redis.Client) *redisStateStorage {
-	return &redisStateStorage{client: client, locker: redislock.New(client)}
+type LoginProgressStorage struct {
```

Comment:
I think it can be simplified even more. I would suggest to use weighted semaphore (`semaphore.NewWeighted`) and `TryAcquire` method.